### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by psykokwak4**
- Added documentation for `YLeafList`.
- Refactored `bundle_translator.py`.
- Deleted `template.json`.

### CODE REVIEW

File: sdk/python/docsgen/types_doc.rst [hunk 1]
```diff
@@ -57,8 +57,21 @@ Example use of YList type
...
+
+- The node being configured (ipv4_dscp) under the ydk.models.asr9k.Cisco_IOS_XR_asr9k_policymgr_cfg module:
+**ipv4_dscp:** An leaflist of Match IPv4 DSCP. **type:** YLeafList
+
+.. code-block:: python
+
+    from ydk.models.asr9k.Cisco_IOS_XR_asr9k_policymgr_cfg import PolicyManager
+
+    match = PolicyManager.ClassMaps.ClassMap.Match()
+    match.ipv4_dscp.extend(['15', '16'])
```
Comment (abhikeshav) [inline sdk/python/docsgen/types_doc.rst]: Looks good. Can you add more examples of the features of YLeafList, including

1. splicing
2. only adding unique elements to leaf-list is allowed

?

Thanks

File: ydkgen/resolver/bundle_translator.py [hunk 2]
```diff
@@ -24,214 +24,163 @@
...
+        logger.debug('Removing folder %s:' % tmp_dir)
+        rmtree(tmp_dir)
+
+def check_envs():
+    if not os.environ.has_key('YDKGEN_HOME'):
+        logger.error('YDKGEN_HOME not set.')
+        print >> sys.stderr, "Need to have YDKGEN_HOME set!"
+        sys.exit(1)
 
+def populate_temple(in_file, out_file):
```
Comment (abhikeshav) [inline ydkgen/resolver/bundle_translator.py]: Type? Should this be populate_template?

File: sdk/python/docsgen/types_doc.rst (hunk 1 above)
Comment (psykokwak4) [inline sdk/python/docsgen/types_doc.rst]: Ok

File: ydkgen/resolver/bundle_translator.py (hunk 2 above)
Comment (psykokwak4) [inline ydkgen/resolver/bundle_translator.py]: Yes,  you're right, and the same with the line at the bottom of this file, sorry about that, will change in next commit.

File: ydkgen/resolver/bundle_resolver.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+            self.modules = []
+            self.dependencies = []
+            if 'modules' in data:
+                for m in data['modules']:
+                    self.modules.append(Module(m))
+            if 'dependencies' in data['bundle']:
+                for d in data['bundle']['dependencies']:
+                    self.dependencies.append(BundleDependency(d))
+        except KeyError as e:
+            raise YdkGenException('Bundle file is not well formatted.')
```
Comment (psykokwak4) [inline ydkgen/resolver/bundle_resolver.py]: well-formed

**Pull request merged**